app/tests/unit/validators/permission.spec.js

```diff
@@ -0,0 +1,186 @@
+        expect(permCode).toEqual(stringMultiModel.describe());
+      });
+    });
+  });
+});
+
+describe('addPermissions', () => {
+
+  describe('params', () => {
+    const params = schema.addPermissions.params.describe();
+
+    describe('objId', () => {
+      const objId = params.keys.objId;
+
+      it('is the expected schema', () => {
+        expect(objId).toEqual(uuidv4.describe());
+      });
+    });
+  });
+
+  describe('body', () => {
+    const body = schema.addPermissions.body.describe();
+
+    it('is an array', () => {
+      expect(body).toBeTruthy();
+      expect(body.type).toEqual('array');
+      expect(Array.isArray(body.items)).toBeTruthy();
+    });
+
+    it('is required', () => {
+      expect(body.flags).toBeTruthy();
+      expect(body.flags).toEqual(expect.objectContaining({presence: 'required'}));
+    });
+
+    it('should contain userId', () => {
+      expect(body.items).toEqual(expect.arrayContaining([
+        expect.objectContaining({
+          keys: expect.objectContaining({
+            userId: expect.objectContaining({
+              type: 'string',
+              flags: expect.objectContaining({presence: 'required'}),
+              rules: expect.arrayContaining([
+                expect.objectContaining({
+                  name: 'guid',
+                  args: {
+                    options: { version: 'uuidv4'}
+                  }
+                })
+              ])
+            })
+          })
+        })
+      ]));
+    });
+
+    it('should contain permCode', () => {
+      expect(body.items).toEqual(expect.arrayContaining([
+        expect.objectContaining({
+          keys: expect.objectContaining({
+            permCode: expect.objectContaining({
+              type: 'string',
+              flags: expect.objectContaining({presence: 'required'}),
+              rules: expect.arrayContaining([
+                expect.objectContaining({
+                  name: 'max',
+                  args: {
+                    limit: 255
+                  }
+                })
+              ])
+            })
+          })
+        })
+      ]));
+    });
+  });
+});
+
+describe('removePermissions', () => {
+
+  describe('params', () => {
+    const params = schema.removePermissions.params.describe();
+
+    describe('objId', () => {
+      const objId = params.keys.objId;
+
+      it('is the expected schema', () => {
+        expect(objId).toEqual(uuidv4.describe());
+      });
+    });
+  });
+
+  describe('query', () => {
+    const query = schema.listPermissions.query.describe();
+
+    describe('userId', () => {
+      const userId = query.keys.userId;
+
+      it('is the expected schema', () => {
+        expect(userId).toEqual(uuidv4MultiModel.describe());
+      });
+    });
+
+    describe('permCode', () => {
+      const permCode = query.keys.permCode;
+
+      it('is the expected schema', () => {
+        expect(permCode).toEqual(stringMultiModel.describe());
+      });
+    });
```

Comment:
I can't quite remember, but permCode may be one of the string values we explicitly define as an enum on the API spec. If that is the case, we could go one step further and do string matches to ensure that only values that are within that enum set pass.